embedchain/utils.py

```diff
@@ -155,6 +155,10 @@ def detect_datatype(source: Any) -> DataType:
             logging.debug(f"Source of `{formatted_source}` detected as `docx`.")
             return DataType.DOCX
 
+        if url.path.endswith(".yaml"):
+            logging.debug(f"Source of `{formatted_source}` detected as `yaml`.")
+            return DataType.OPENAPI
```

Comment:
I am not sure if we want to detect all the `*.yaml` files as openapi data type.

Reply:
True, adding the safeguard for openapi type yaml file would be a good idea.